### PR TITLE
More ODE Solvers

### PR DESCRIPTION
**Pull request by jaketimothy**
This is a rework of the ode solvers.  This makes available all of the adaptive step size algorithms included in Apache commons-math3 until native scala implementations are developed, if desired.

Addresses #478.

### CODE REVIEW

File: math/src/main/scala/breeze/integrate/ApacheAdaptiveStepIntegrator.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  override def integrate(
+    f: (DenseVector[Double], Double) => DenseVector[Double],
+    y0: DenseVector[Double],
+    t: Array[Double]): Array[DenseVector[Double]] = {
+
+    // If error tolerances are not specified, fill with default.
+    val relativeTolerance = someRelTol getOrElse DenseVector.fill(y0.length)(ApacheAdaptiveStepIntegrator.defaultRelTol)
+    val absoluteTolerance = someAbsTol getOrElse DenseVector.fill(y0.length)(ApacheAdaptiveStepIntegrator.defaultAbsTol)
+
+    inner.setStepSizeControl(minStep, maxStep, absoluteTolerance.toArray, relativeTolerance.toArray)
```
Comment (dlwh) [inline math/src/main/scala/breeze/integrate/ApacheAdaptiveStepIntegrator.scala]: i'd rather we make this threadsafe

**Comment by dlwh** [comment]
other than the thread safety issue, basically looks good!

**Comment by dlwh** [comment]
sorry, totally didn't notice this got updated (I don't get email notifications when someone updates a PR, I don't know if that's possible...)

**Comment by jaketimothy** [comment]
Hey, don't worry about it!  In the future I'll comment when I'm done with an update.

**Pull request merged**